python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1380 @@
+                TEST = x[:, np.newaxis]
+                slope, _, _, _ = np.linalg.lstsq(TEST, y)
+                slope = slope[0]
+                res = y - slope * x
+                resMean = afwMath.makeStatistics(res, afwMath.MEANCLIP, sctrl).getValue()
+                resStd = np.sqrt(afwMath.makeStatistics(res, afwMath.VARIANCECLIP, sctrl).getValue())
+                # xxx check this line performs the same
+                index = np.where((res > (resMean+nSigmaClip*resStd)) | (res < (resMean-nSigmaClip*resStd)))
+                self.log.debug("%.3f %.3f %.3f %.3f" % (resMean, resStd, np.max(res), nSigmaClip))
+                if np.shape(np.where(index))[1] == 0 or (nIter >= maxIter):  # run out of points or iters
+                    break
+                x = np.delete(x, index)
+                y = np.delete(y, index)
+
+            return slope, 0
+
+        while nIter < maxIter:
+            nIter += 1
+            self.log.debug("Iteration # %s using %s elements:" % (nIter, np.shape(x)[0]))
+            xx = np.vstack([x, np.ones(len(x))]).T
+            ret, _, _, _ = np.linalg.lstsq(xx, y)
+            slope, intercept = ret
+            res = y - slope*x - intercept
+            resMean = afwMath.makeStatistics(res, afwMath.MEANCLIP, sctrl).getValue()
+            resStd = np.sqrt(afwMath.makeStatistics(res, afwMath.VARIANCECLIP, sctrl).getValue())
+            index = np.where((res > (resMean + nSigmaClip * resStd)) | (res < resMean - nSigmaClip * resStd))
+            self.log.debug("%.3f %.3f %.3f %.3f" % (resMean, resStd, np.max(res), nSigmaClip))
+            if np.shape(np.where(index))[1] == 0 or (nIter >= maxIter):  # run out of points, or iterations
+                break
+            x = np.delete(x, index)
+            y = np.delete(y, index)
+
+        return slope, intercept
+
+    def generateKernel(self, corrs, means, objId, rejectLevel=None):
+        """Generate the full kernel from a list of cross-correlations and means.
+
+        Taking a list of quarter-image, gain-corrected cross-correlations,
+        do a pixel-wise sigma-clipped mean of each,
+        and tile into the full-sized kernel image.
+
+        Each corr in corrs is one quarter of the full cross-correlation,
+        and has been gain-corrected. Each mean in means is a tuple of the means
+        of the two individual images, corresponding to that corr.
+
+        Parameters:
+        -----------
+        corrs : `list` of `numpy.ndarray`, (Ny, Nx)
+            A list of the quarter-image cross-correlations
+        means : `dict` of `tuples` of `floats`
+            The means of the input images for each corr in corrs
+        rejectLevel : `float`, optional
+            This is essentially is a sanity check parameter.
+            If this condition is violated there is something unexpected
+            going on in the image, and it is discarded from the stack before
+            the clipped-mean is calculated.
```

Comment:
Not sure, but I think you just need to scroll down. I see:
```        rejectLevel : `float`, optional
            This is essentially is a sanity check parameter.
            If this condition is violated there is something unexpected
            going on in the image, and it is discarded from the stack before
            the clipped-mean is calculated.
            If not provided, a default value of 0.2 is picked up from the
            task config.```

Reply:
I suggest: If not provided then config.xcorrCheckRejectLevel is used. It is risky to document the default value of a config parameter anywhere (except in the config itself, where it is set) because it may change, and of course obs packages may override it.